contracts/pcv/morpho/MorphoCompoundPCVDeposit.sol

```diff
@@ -66,54 +129,256 @@ contract MorphoCompoundPCVDeposit is PCVDeposit {
         return token;
     }
 
+    /// ------------------------------------------
+    /// ----------- Permissionless API -----------
+    /// ------------------------------------------
+
     /// @notice deposit ERC-20 tokens to Morpho-Compound
-    function deposit() public whenNotPaused {
+    /// non-reentrant to block malicious reentrant state changes
+    /// to the lastRecordedBalance variable
+    function deposit() public whenNotPaused nonReentrant {
+        /// ------ Check ------
+
         uint256 amount = IERC20(token).balanceOf(address(this));
         if (amount == 0) {
             /// no op to prevent revert on empty deposit
             return;
         }
 
-        IERC20(token).approve(address(MORPHO), amount);
-        MORPHO.supply(
+        int256 startingRecordedBalance = lastRecordedBalance.toInt256();
+
+        /// ------ Effects ------
+
+        /// compute profit from interest accrued and emit an event
+        /// if any profits or losses are realized
+        _recordPNL();
+
+        /// increment tracked recorded amount
+        /// this will be off by a hair, after a single block
+        /// negative delta turns to positive delta (assuming no loss).
+        lastRecordedBalance += amount;
+
+        /// ------ Interactions ------
+
+        IERC20(token).approve(address(morpho), amount);
+        IMorpho(morpho).supply(
             cToken, /// cToken to supply liquidity to
             address(this), /// the address of the user you want to supply on behalf of
             amount
         );
 
+        int256 endingRecordedBalance = balance().toInt256();
+
+        if (pcvOracle != address(0)) {
+            IPCVOracle(pcvOracle).updateLiquidBalance(
+                endingRecordedBalance - startingRecordedBalance
+            );
+        }
```

Comment:
This piece of code is written 4 times and could have been in an internal function

Reply:
We could refactor this into a helper function, but it's too late in the game to change code before a deployment.